### ISSUE TITLE
taxonomy files' header

### ISSUE THREAD

**Issue by lnnoemi** [clarin-eric/ParlaMint#431]
@matyaskopp, I try to follow the structure of the taxonomy files in the Czech sample data folder. However, if I submit my files with these two lines at the top:
`<?xml version="1.0" ?>
<taxonomy xmlns="http://www.tei-c.org/ns/1.0" xml:id="subcorpus">`
the validation of the schema fails, and I get the error message as `element "taxonomy" not allowed here; expected element "teiCorpus"`
Could you, please, help me how to correctly write these files?

**Comment by TomazErjavec** [comment]
The start of the file is ok, but it looks like the wrong schema is used. Is the name of the file "ParlaMint-taxonomy-subcorpus.xml"?
If so, maybe the validation script has a bug.

**Comment by matyaskopp** [comment]
just run
```bash
make factorize-teiHeader-INPLACE-HU
```
in your repository (it is modifying corpus root files in place)

 add newly generated files into git (taxonomies, listOrg and listPerson) and commit changes

**Comment by lnnoemi** [comment]
Ok, thanks, I've just did it, I hope it will be fine now.

**Comment by TomazErjavec** [comment]
If it was fine, @lnnoemi, can you close this issue?

**Comment by hpreki** [comment]
@lnnoemi - could you confirm, that this _really_ solve your validation problem? (and if not: please re-open the issue)
I do have the exactly same issue, and `make factorize-teiHeader-...` does not help at all.
 @matyaskopp @TomazErjavec  I also do not understand how it _could_ help:  my `ParlaMint-AT.xml` starts out with include-statements for  all taxonomies and lists - i.e. it _is_ in a "fully factorized"-format already. 
 It is _created_ like that, i.e. it has from the beginning on,  the same _factorized_ structure as  `Data/ParlaMint-CZ/` 

Therefore it is not surprising that: 
*  ` make val-schema-CZ` shows the same errors which @lnnoemi reported:
```
make val-schema-CZ
find Data -type f -path "Data/ParlaMint-CZ/ParlaMint-*.xml" | grep -v '.ana.' | grep -v '_' | xargs java -jar /usr/share/java/jing.jar Schema/ParlaMint-teiCorpus.rng 	
Data/ParlaMint-CZ/ParlaMint-CZ-listOrg.xml:2:46: error: element "listOrg" not allowed here; expected element "teiCorpus"
Data/ParlaMint-CZ/ParlaMint-CZ-listPerson.xml:2:49: error: element "listPerson" not allowed here; expected element "teiCorpus"
Data/ParlaMint-CZ/ParlaMint-CZ-taxonomy-NER.cnec2.0.xml:2:68: error: element "taxonomy" not allowed here; expected element "teiCorpus"
Data/ParlaMint-CZ/ParlaMint-CZ-taxonomy-meeting.parts.xml:2:70: error: element "taxonomy" not allowed here; expected element "teiCorpus"
...
Data/ParlaMint-CZ/ParlaMint-taxonomy-parla.legislature.xml:2:74: error: element "taxonomy" not allowed here; expected element "teiCorpus"
Data/ParlaMint-CZ/ParlaMint-taxonomy-subcorpus.xml:2:66: error: element "taxonomy" not allowed here; expected element "teiCorpus"
```

* `make factorize-teiHeader-CZ` / `make factorize-teiHeader-AT` will do not anything useful  

_My_ guess is: the schema-validator tries to validate each file mentioned in a `<xi:include>` against the `Schema/ParlaMint-teiCorpus.rng` _before_ it is included. And the stand-alone taxonomy of course is not a valid ParlaMint-TEI-file.


Any solution? 
* de-factorize files? i.e. have _no_ external taxonomies and personLists at validation-time (doable, but then we still have a (factorized) end-product which will not validate?
* enforce validation only _after_ external files have been included? Should work for the small sample, but is  probably  "catastrophic" for the full-corpus because also all TEI-files would be included prior to validation?  
* ???

**Comment by lnnoemi** [comment]
It solved it in a way that I did not run make val-schema-HU (or other validation scripts) afterwards, just uploaded the sample, as @matyaskopp suggested. But the error message did not disappear.

**Comment by TomazErjavec** [comment]
I think this is a bug in the validation script - the taxonomies, listPerson, and listOrg now have their own RelaxNG schemas, and they should be validated against these schemas, not against teiCorpus.rng. At the same time, teiCorpus.rng has been changed to allow either the non-factorized or factorized variants.

**Comment by lnnoemi** [comment]
I thought that the solution, at least for now, is to ignore this message during the local validation, as the sample I uploaded was smoothly validated after the commit.

**Comment by matyaskopp** [comment]
> * `make factorize-teiHeader-CZ` / `make factorize-teiHeader-AT` will do not anything useful

it is placing the factorized result into `Data/ParlaMint-XX/factorize-teiHeader` subdirectory so that you can check it.

**Comment by hpreki** [comment]
> it is placing the factorized result into `Data/ParlaMint-XX/factorize-teiHeader` subdirectory so that you can check it.

In _my_ tests `Data/ParlaMint-XX/factorize-teiHeader` will only contain the  unchanged copies of ` ParlaMint-CZ.ana.xml
ParlaMint-CZ.xml` and `ParlaMint-CZ.ana.xml`: it is fair that factorization does nothing when everything already is factorized.

**Comment by hpreki** [comment]
Thanks a lot  to all of you for your lightning-speed support! I happily will ignore the validation-error and will proceed with `make validate-parlamint-AT` ;-)

**Comment by matyaskopp** [comment]
fixed with: 37b1a8a0e726bab525a13d46251b2ae04f307978
and merged into data branch
